Dashboard/Dashboard/Presentation/Elements/PrimaryCardView.swift

```diff
@@ -234,15 +243,25 @@ public struct PrimaryCardView: View {
                 .font(Theme.Fonts.titleLarge)
                 .foregroundStyle(Theme.Colors.textPrimary)
                 .lineLimit(3)
-            if let courseEndDate {
-                Text(courseEndDate.dateToString(style: .courseEndsMonthDDYear))
-                    .font(Theme.Fonts.labelMedium)
-                    .foregroundStyle(Theme.Colors.textSecondaryLight)
-            } else if let courseStartDate {
-                Text(courseStartDate.dateToString(style: .courseStartsMonthDDYear))
-                    .font(Theme.Fonts.labelMedium)
-                    .foregroundStyle(Theme.Colors.textSecondaryLight)
+           
+            if courseStartDate != nil || courseEndDate != nil || auditAccessExpires != nil {
+                CourseAccessMessageView(
+                    startDate: courseStartDate,
+                    endDate: courseEndDate,
+                    auditAccessExpires: auditAccessExpires,
+                    startDisplay: startDisplay,
+                    startType: startType
+                )
             }
+//            if let courseEndDate {
```

Comment:
Are you sure that we don't need that code?

Reply:
`CourseAccessMessageView` is doing the job now. Removed the commented code.